### PR TITLE
activity: we shouldn't operate on events we have no indices for

### PR DESCRIPTION
**Pull request by arthyn**
OTT, this was causing crashes when scrolling in the activity feed. Fixes TLON-2436

PR Checklist
- [X] Includes changes to desk files
- [ ] Describes how you tested the PR locally (test ship vs livenet)
- [ ] If a new feature, includes automated tests
- [ ] Comments added anywhere logic may be confusing without context

### CODE REVIEW

**Review by patosullivan — APPROVED** [review]
Makes sense to me

File: desk/app/activity.hoon [hunk 1]
```diff
@@ -476,6 +476,7 @@
   ?:  (gth time real-start)  [~ | acc]
   :-  ~   :-  |
   =/  =source:a  (source:evt -.event)
+  ?.  (~(has by indices) source)  [~ | acc]
```
Comment (Fang-) [inline desk/app/activity.hoon:479]: This `mint-nice`s and shouldn't be passing CI. We need to bring backend build checks back.

```suggestion
  ?:  (gth time real-start)  [~ | acc]
  :-  ~   :-  |
  =/  =source:a  (source:evt -.event)
  ?.  (~(has by indices) source)  acc
```

Or, more invasively:

```suggestion
  :+  ~  |
  ?:  (gth time real-start)  acc
  =/  =source:a  (source:evt -.event)
  ?.  (~(has by indices) source)  acc
```

**Comment by arthyn** [comment]
@Fang- yeah we should, sorry I thought I built this successfully.

**Review by Fang- — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**